File: Server/src/gs/beans/chuhan/gsp/RoleInfo.java

package chuhan.gsp;

import com.goldhuman.Common.Marshal.Marshal;
import com.goldhuman.Common.Marshal.OctetsStream;
import com.goldhuman.Common.Marshal.MarshalException;

public class RoleInfo implements Marshal {
	public long roleid; // ID
	public java.lang.String rolename; // 名称
	public short rolelevel; // 等级

	public RoleInfo() {
		rolename = "";
	}

	public RoleInfo(long _roleid_, java.lang.String _rolename_, short _rolelevel_) {
		this.roleid = _roleid_;
		this.rolename = _rolename_;
		this.rolelevel = _rolelevel_;
	}

	public final boolean _validator_() {
		return true;
	}

	public OctetsStream marshal(OctetsStream _os_) {
		_os_.marshal(roleid);
		_os_.marshal(rolename, "UTF-16LE");
		_os_.marshal(rolelevel);
		return _os_;
	}

	public OctetsStream unmarshal(OctetsStream _os_) throws MarshalException {
		roleid = _os_.unmarshal_long();
		rolename = _os_.unmarshal_String("UTF-16LE");
		rolelevel = _os_.unmarshal_short();
		return _os_;
	}

	public boolean equals(Object _o1_) {
		if (_o1_ == this) return true;
		if (_o1_ instanceof RoleInfo) {
			RoleInfo _o_ = (RoleInfo)_o1_;
			if (roleid != _o_.roleid) return false;
			if (!rolename.equals(_o_.rolename)) return false;
			if (rolelevel != _o_.rolelevel) return false;
			return true;
		}
		return false;
	}

	public int hashCode() {
		int _h_ = 0;
		_h_ += (int)roleid;
		_h_ += rolename.hashCode();
		_h_ += rolelevel;
		return _h_;
	}

	public String toString() {
		StringBuilder _sb_ = new StringBuilder();
		_sb_.append("(");
		_sb_.append(roleid).append(",");
		_sb_.append("T").append(rolename.length()).append(",");
		_sb_.append(rolelevel).append(",");
		_sb_.append(")");
		return _sb_.toString();
	}

}

File: client/main/Assets/Program/Scripts/Msg/beans/gsp/RoleInfo.cs
using System;
namespace GNET
{
    public class RoleInfo : Marshal
	{
        public long roleid; // ID
        public string rolename; // Ãû³Æ
        public short rolelevel; // µÈ¼¶

        public RoleInfo()
        {
            rolename = "";
        }

        public RoleInfo(long _roleid_, string _rolename_, short _rolelevel_)
        {
            this.roleid = _roleid_;
            this.rolename = _rolename_;
            this.rolelevel = _rolelevel_;
        }

		public override OctetsStream marshal(OctetsStream os)
		{
            os.marshal(roleid);
            os.marshal(rolename);
            os.marshal(rolelevel);
			return os;
		}

		public override OctetsStream unmarshal(OctetsStream os)
		{
            roleid = os.unmarshal_long();
            rolename = os.unmarshal_String();
            rolelevel = os.unmarshal_short();
			return os;
		}

	}
}
